### PR TITLE
AdGuard Home: New role

### PR DESCRIPTION
**Pull request by sevospl**
# Description

[AdGuard Home](https://hub.docker.com/r/adguard/adguardhome) is a network-wide, open source software for blocking ads & tracking and for gaining control over all traffic in your home network. After you set it up, it'll cover ALL devices in your home Wi-Fi network, and you won't need any client-side software for that. At the same time, it provides a user-friendly web interface that allows you to easily manage the traffic, even from a mobile device.

# How Has This Been Tested?

I tested the role in wild and everything worked as expected.

# New Role Checklist:

- [✔️] <not ready yet> I have reviewed the [Contribute page in the wiki](https://docs.saltbox.dev/community/basics/#contributing-to-community-apps)
- [✔️] I have updated the header in any files I may have used as templates with my own information
- [✔️] I have added my new role to `community.yml`
- [✔️] I have verified that any Docker images used are current and supported.
- [✔️] I have made corresponding changes to the documentation or have submitted a separate PR to the docs repo for this purpose.

### CODE REVIEW

**Comment by maximuskowalski** [comment]
I'm not sure that running a DNS server is safe or belongs anywhere but within a home network. The general consensus seems to be that if you do try to do this you should run it with a split VPN. Same goes with pi-hole. Saltbox is able to run from within the home network so maybe if this gets merged it should be made clear the dangers involved.

**Comment by saltydk** [comment]
Please try to submit clean PRs. Contains remnants from the previous PR you submitted.

File: roles/adguardhome/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+adguardhome_docker_commands_custom: []
+adguardhome_docker_commands: "{{ adguardhome_docker_commands_default
+                            + adguardhome_docker_commands_custom }}"
+
+# Volumes
+adguardhome_docker_volumes_default:
+  - "{{ adguardhome_paths_location }}/work:/opt/adguardhome/work"
+  - "{{ adguardhome_paths_location }}/conf:/opt/adguardhome/conf"
+adguardhome_docker_volumes_custom: []
+bazarr_docker_volumes_theme: []
```
Comment (saltydk) [inline roles/adguardhome/defaults/main.yml:102]: Oversight?

**Comment by saltydk** [comment]
I am of a similar opinion as Max regarding the rabbit-hole this addition might lead us down. While an initial search does suggest that it might be more secure to run publicly as opposed to PiHole.

Hopefully someone knows more about this than me and can address such concerns. @owine @chazlarson maybe?

**Comment by owine** [comment]
> I am of a similar opinion as Max regarding the rabbit-hole this addition might lead us down. While an initial search does suggest that it might be more secure to run publicly as opposed to PiHole.
> 
> 
> 
> Hopefully someone knows more about this than me and can address such concerns. @owine @chazlarson maybe?

I think this is valuable to have in the repo as it's something people want if using a LAN server, but unless there is a way to configure by default local access only, we should not implement.

**Comment by sevospl** [comment]
> Oversight?

@saltydk Nice eyes - I'll fix it :)

The current setup has only DoH (DNS-over-HTTPS) enabled and additionally AdGuard Home allows you set a whitelist.

DNS amplification attacks cannot be performed against DoT/DoH - they use TCP which doesn't offer the same amplification as UDP[1].

[1] https://www.reddit.com/r/networking/comments/izyokk/can_dns_amplification_attacks_be_performed_on_dot/

**Comment by saltydk** [comment]
Still waiting for a clean PR btw.

**Comment by maximuskowalski** [comment]
Do you want me to fix formatting and test this?

**Comment by saltydk** [comment]
Could we have this moved to the new sandbox repo? Stuff I personally don't plan to maintain will exist there.

**Comment by saltydk** [comment]
closed in favor of https://github.com/saltyorg/Sandbox/pull/3

**Pull request closed without merging**